### PR TITLE
fixes #794 - add option to bind / unbind committee meetings from bills

### LINKED ISSUE #794: אופציה לקשר ישיבות הוועדה, לוועדות אחרי קריאה ראשונה
לא קיימת אופציה לקישר

### PR DESCRIPTION
**Pull request by OriHoch**
#### Relating issues 
* #794

### CODE REVIEW

**Comment by OriHoch** [comment]
forgot translations.. adding now

**Comment by OriHoch** [comment]
I added the translations changes but there are a lot of changes, I only added these strings:

#: templates/laws/bill_unbind_committee_meeting.html:4
msgid "Unbind committee meeting"
msgstr "הסרת ישיבת וועדה"

#: templates/laws/bill_detail.html:322 templates/laws/bill_detail.html:428
msgid "Suggest a committee meeting"
msgstr "הוספת ישיבת ועדה"


#: templates/laws/bill_detail.html:329 templates/laws/bill_detail.html:435
msgid ""
"Start typing a committee meeting title or date and choose one from the list"
msgstr ""
"יש להקליד כותרת או תאריך ישיבת ועדה, ולבחור את הישיבה המתאימה מתוך התוצאות"

File: laws/views.py [hunk 1]
```diff
@@ -533,6 +540,34 @@ def bill_unbind_vote(request, object_id, vote_id):
...
         return render_to_response("laws/bill_unbind_vote.html", context)
 
+@login_required
+def bill_unbind_committee_meeting(request, object_id, cm_id, cm_stage):
+    try:
+        bill = Bill.objects.get(pk=object_id)
+        cm = CommitteeMeeting.objects.get(pk=cm_id)
+    except ObjectDoesNotExist:
+        raise Http404
+    if request.method == 'POST':  # actually unbind
```
Comment (alonisser) [inline laws/views.py]: Why not DELETE? isn't this an ajax call we control?

File: laws/views.py [hunk 2]
```diff
@@ -873,6 +908,30 @@ def knesset_proposal_auto_complete(request):
...
     return HttpResponse(json.dumps(result), mimetype='application/json')
 
+@require_http_methods(["GET"])
+def committee_meeting_auto_complete(request):
+    if not request.GET.get('query'):
+        raise Http404
+
+    q = request.GET['query']
+
+    options = CommitteeMeeting.objects.filter(
```
Comment (alonisser) [inline laws/views.py]: Maybe rename to "optional_committee_meetings" for clarify?

File: laws/views.py [hunk 3]
```diff
@@ -873,6 +908,30 @@ def knesset_proposal_auto_complete(request):
...
+        Q(date_string__icontains=q) | Q(topics__icontains=q)
+    )[:30]
```
Comment (alonisser) [inline laws/views.py]: Why [:30] seems granted to miss some, and we have no option in UI to fix. unless there is a "more committees" options integrated with scroll down behavior

File: laws/views.py [hunk 4]
```diff
@@ -873,6 +908,30 @@ def knesset_proposal_auto_complete(request):
...
+
+    data = []
+    suggestions = []
+    for i in options:
```
Comment (alonisser) [inline laws/views.py]: maybe rename i for meeting, for clarity

File: laws/views.py [hunk 5]
```diff
@@ -873,6 +908,30 @@ def knesset_proposal_auto_complete(request):
...
+        title = u'{0} - {1}'.format(i.date_string, i.topics)
+        data.append(i.id)
+        suggestions.append(title)
+
+    result = {'query': request.GET['query'],
+              'suggestions': suggestions,
+              'data': data}
+
+    return HttpResponse(json.dumps(result), mimetype='application/json')
```
Comment (alonisser) [inline laws/views.py]: There is  a built in JsonResponse in Django that takes care of dump and mimetype exactly for that

File: laws/views.py (hunk 1 above)
Comment (OriHoch) [inline laws/views.py]: not to my knowledge.. it's a regular form submit, not ajax

File: laws/views.py (hunk 5 above)
Comment (OriHoch) [inline laws/views.py]: unfortunately not in v1.6 ..

should be easy to backport it but that's a different issue I guess

**Comment by OriHoch** [comment]
@alonisser - fixed according to your comments + added tests

**Pull request merged**